pkg/roachpb/api.proto

```diff
@@ -2528,6 +2543,12 @@ message Header {
 
   util.tracing.tracingpb.TraceInfo trace_info = 25;
 
+  // This "any type" field is a serialized descpb.IndexFetchSpec that can be
+  // used to initialize a columnar scan for MVCC. We use this generic type to
+  // resolve some painful circular dependency issues.
+  // TODO: better comment.
+  google.protobuf.Any index_fetch_spec = 29;
```

Comment:
Have you considered making this bytes or rather string? I think that a lot of the time the index_fetch_spec is going to be the same. I imagine nobody is every going to use the protobuf directly but rather will build some other sort of decoder struct from it. I imagine we'll want to cache those things. The string itself can be a hash key into the cache. I think it'd be nice to avoid decoding and parsing the specs every time. Ideally we'd define some type like `type EncodedFetchSpec string` and use a cast type.

Reply:
No, I haven't really considered this, but thanks for taking a look.

I'm currently trying to make the prototype work without too much thought to this kind of questions - at the moment, I'm struggling with all the intricacies of advancing the iterator in `getAndAdvance`, especially in the presence of multiple versions of the same keys. That code is hard to comprehend :/